### PR TITLE
3109 lakectl ingest tests

### PR DESCRIPTION
**Pull request by itai-david**
Verification tests for `lakectl ingest` command output format

### CODE REVIEW

File: nessie/lakectl_test.go [hunk 1]
```diff
@@ -295,3 +301,52 @@ func TestLakectlAuthUsers(t *testing.T) {
...
+	// location. This is done as the `lakectl ingest` commad runs with local aws settings and can only access the
+	// corresponding region
+
+	s3Client, err := store.GetS3Client()
+	require.NoError(t, err, "Failed to initialize S3 session")
+
+	_, err = s3Client.CreateBucket(&s3.CreateBucketInput{Bucket: aws.String(tempBucketName)})
+	require.NoError(t, err, "Failed to create bucket", tempBucketName)
+
+	defer cleanS3Bucket(s3Client, tempBucketName)
```
Comment (itaiad200) [inline nessie/lakectl_test.go:325]: Creating a bucket per run can blow up if deleting the bucket doesn't work as expected, or the test is stopped half way through.

I suggest instead to use a predefined bucket and use prefixes for the ingestion and the repo part.
For example `s3://test-bucket-for-lakectl-ingestion/` where the prefix ingestion `ingestion-raw-run-xxxxx` and storage namespace for the matching repo `ingestion-repo-run-xxxxx`. Then this bucket can have lifecycle policies for cleanups (let's say 2 weeks) and you don't have to worry about dangling buckets.

You can also use the bucket `s3://nessie-system-testing-data` for the raw to-be-ingested data. That bucket data isn't deleted and you won't have to upload these files yourself.

File: nessie/lakectl_test.go [hunk 2]
```diff
@@ -295,3 +301,52 @@ func TestLakectlAuthUsers(t *testing.T) {
 	// Cleanup
 	RunCmdAndVerifySuccess(t, Lakectl()+" auth users delete --id "+userName, false, "User deleted successfully\n", vars)
 }
+
+func TestLakectlIngest(t *testing.T) {
```
Comment (itaiad200) [inline nessie/lakectl_test.go:305]: This test can't work with Azure and GCP.

**Review by itaiad200 — CHANGES_REQUESTED** [review]
Thanks for this much needed tests!

I do think that you should use predefined bucket and data, instead of creating buckets for each tests. Also - you need to disable the test until we support ingest for gcp and azure.

**Review by N-o-Z — APPROVED** [review]
For me it looks good, I don't really have anything to add to what @itaiad200 already mentioned

File: nessie/lakectl_test.go (hunk 2 above)
Comment (itai-david) [reply nessie/lakectl_test.go]: Correct
Added the adapter type to nessie via env var. Test is skipped if adapter is not S3

File: .github/workflows/nessie.yaml [hunk 3]
```diff
@@ -724,6 +727,7 @@ jobs:
           LAKEFS_BLOCKSTORE_TYPE: azure
           LAKEFS_BLOCKSTORE_AZURE_STORAGE_ACCOUNT: ${{ secrets.LAKEFS_BLOCKSTORE_AZURE_STORAGE_ACCOUNT }}
           LAKEFS_BLOCKSTORE_AZURE_STORAGE_ACCESS_KEY: ${{ secrets.LAKEFS_BLOCKSTORE_AZURE_STORAGE_ACCESS_KEY }}
+          NESSIE_BLOCKSTORE_TYPE: azure
```
Comment (itaiad200) [inline .github/workflows/nessie.yaml:730]: Can this be different than `LAKEFS_BLOCKSTORE_TYPE`? Why not use the same env var?

**Review by itaiad200 — APPROVED** [review]
Approved with a single minor comment, thx!

File: .github/workflows/nessie.yaml (hunk 3 above)
Comment (itai-david) [reply .github/workflows/nessie.yaml:730]: It is used inside `nessiecontext, and `viper` only reads variables with `NESSIE_` prefix

**Pull request merged**